### PR TITLE
fixes bug 1271621 - Add reason foreign key to correlations table

### PR DESCRIPTION
**Pull request by peterbe**

### CODE REVIEW

**Comment by peterbe** [comment]
I was able to run this locally:
```
 PYTHONPATH=. alembic -c config/alembic.ini upgrade head
```
and it did add the column to my local PG.

**Comment by adngdb** [comment]
I believe you forgot to update the associated model.

**Comment by peterbe** [comment]
@adngdb Fixed now.

File: alembic/versions/a44fb7695c10_add_reasons_foreign_key_to_correlations.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            'date',
+            'signature_id',
+            'reason_id',
+        ],
+        unique=True,
+    )
+
+
+def downgrade():
+    op.drop_column(
```
Comment (adngdb) [inline alembic/versions/a44fb7695c10_add_reasons_foreign_key_to_correlations.py]: Don't you need to drop the index here as well? Or will it get removed automatically when removing one of its columns?

Comment (peterbe) [inline alembic/versions/a44fb7695c10_add_reasons_foreign_key_to_correlations.py]: That's what I had but it broke the tests. I'm trying to comprehend why.

**Pull request closed without merging**